docs/manual-testing-playbook.md

```diff
@@ -0,0 +1,997 @@
+lxc exec tailscale-1 -- tailscale drive share mydrive snap/tailscale/common/drive
+lxc exec tailscale-1 -- tailscale drive list  # mydrive listed
+lxc exec tailscale-1 -- tailscale drive rename mydrive mydrive2
+lxc exec tailscale-1 -- tailscale drive list  # mydrive2 listed
+lxc exec tailscale-1 -- tailscale drive unshare mydrive2
+lxc exec tailscale-1 -- tailscale drive list  # no drives listed
+```
+
+For an explanation of why sharing a drive doesn't work in the confined snap:
+
+See
+https://github.com/tailscale/tailscale/blob/e3c6ca43d3e3cad27714d07b3a9ec20141c9c65c/drive/driveimpl/remote_impl.go#L336-L355 .
+Tailscaled in the snap is running as root, without access to su.
+So the su check fails, and the check for running as non-root user also fails.
+This is regardless of file paths and file permissions.
+You can see the error messages in the tailscaled logs (`snap logs tailscale`).
+
+### Tailscale dns
+
+Tailscale manages the system DNS by default (it can be turned off via `tailscale set --accept-dns=false`).
+It also provides a `tailscale dns` subcommand that can be used to test the resolver and view the status.
+
+Check the status of the Tailscale DNS service on the host:
+
+```bash
+lxc exec tailscale-1 -- tailscale dns status
+```
+
+With the default configuration, this should indicate that Tailscale DNS is enabled, and print information about MagicDNS, Tailscale FQDN (fully qualified domain name) of the machine, DNS routes, and search domains.
+You may see a note about "reading the system DNS configuration is not supported on this platform";
+this does not appear to be related to snap confinement, merely a system limitation.
+
+Now you can test resolving the Tailscale MagicDNS FQDN through the `tailscale dns query` subcommand,
+as well as the system resolver.
+
+```bash
+TAILNET_DOMAIN=$(lxc exec tailscale-1 -- sh -c "tailscale dns status | awk 'match(\$0, /\w+\.ts\.net/) {print substr(\$0, RSTART, RLENGTH); exit}'")
+lxc exec tailscale-1 -- tailscale dns query tailscale-2.$TAILNET_DOMAIN a
+lxc exec tailscale-1 -- dig tailscale-2.$TAILNET_DOMAIN
+```
+
+Both of the above queries should return the same ipv4 address: the tailnet address of `TAILSCALE_VM_2`.
+Resolvectl should confirm the system DNS settings applied by tailscale:
+
+```bash
+lxc exec tailscale-1 -- resolvectl
+```
+
+Note that DNS queries for the bare machine name don't work (no result returned):
+
+```bash
+lxc exec tailscale-1 -- tailscale dns query tailscale-2 a
+lxc exec tailscale-1 -- dig tailscale-2
+```
+
+This is unexpected, but does not appear to be related to snap confinement.
+
+### Tailscale SSH
+
+- Tailscaled in ssh mode (ie. run with `--ssh`) does not work (incoming ssh connections on the tailnet will not work, failing with permission denied).
+- `tailscale ssh` will work if sshing to a host running tailscaled, if that tailscaled is not installed via the strictly confined snap.
+
+### Tailscale update
```

Comment:
Do we want to test `snap refresh` ?

Reply:
Good point, yes that would be good to test.

In this section, I wanted to demonstrate how the `tailscale update` command would behaviour in the snap confinement. Maybe we can add a new section later, or a new document to cover testing snap refresh (and anything else in the scope of snap actions, such as managing tailscaled (the daemon) or viewing the logs).

Reply:
I created https://github.com/canonical/tailscale-snap/issues/38 to track this. :)